<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>RunTerraWorkflow: Run workflows implemented in Terra • RunTerraWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="RunTerraWorkflow: Run workflows implemented in Terra">
<meta property="og:description" content="RunTerraWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RunTerraWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bioBakery.html">Run bioBakery</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/shbrief/RunTerraWorkflow" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>RunTerraWorkflow: Run workflows implemented in Terra</h1>
                        <h4 data-toc-skip class="author">Sehyun Oh</h4>
            
            <h4 data-toc-skip class="date">April 13, 2022</h4>
      
      
      <div class="hidden name"><code>bioBakery.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>For R users with the limited computing resources, we introduce the RunTerraWorkflow package. This package allows users to run workflows implemented in <a href="https://app.terra.bio/#" class="external-link">Terra</a>, a cloud-based genomics platform, without writing any workflow, installing software, or managing cloud resources. Terra’s computing resources rely on Google Cloud Platform (GCP) and to use RunTerraWorkflow, you only need to setup the Terra account once at the beginning.</p>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"RunTerraWorkflow"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
    <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"shbrief/RunTerraWorkflow"</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"biobakeR"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
    <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"shbrief/biobakeR"</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"AnVIL"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"AnVIL"</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RunTerraWorkflow</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnVIL</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-terra-account">Create Terra account<a class="anchor" aria-label="anchor" href="#create-terra-account"></a>
</h3>
<p>You need <a href="https://support.terra.bio/hc/en-us/articles/360034677651-Account-setup-and-exploring-Terra" class="external-link">Terra account setup</a>. Once you have your own Terra account, you need two pieces of information from it to use RunTerraWorkflow package: 1) the email address linked to your Terra account and 2) your billing project name. Here are the basic input parameters used in this vignette. <strong>Modify these with YOUR account information!</strong></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">accountEmail</span> <span class="op">&lt;-</span> <span class="st">"YOUR_EMAIL@gmail.com"</span>
<span class="va">billingProjectName</span> <span class="op">&lt;-</span> <span class="st">"YOUR_BILLING_ACCOUNT"</span>

<span class="fu"><a href="../reference/setCloudEnv.html">setCloudEnv</a></span><span class="op">(</span>accountEmail <span class="op">=</span> <span class="va">accountEmail</span>, 
            billingProjectName <span class="op">=</span> <span class="va">billingProjectName</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="major-steps">Major steps<a class="anchor" aria-label="anchor" href="#major-steps"></a>
</h3>
<p>Here is the table of major functions for three workflow steps - prepare, run, and check result.</p>
<table class="table">
<thead><tr class="header">
<th>Steps</th>
<th>Functions</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Prepare</td>
<td><code>cloneWorkspace</code></td>
<td>copy the template workspace</td>
</tr>
<tr class="even">
<td></td>
<td><code>updateInput</code></td>
<td>take user’s inputs</td>
</tr>
<tr class="odd">
<td>Run</td>
<td><code>launchWorkflow</code></td>
<td>launch the workflow in Terra</td>
</tr>
<tr class="even">
<td></td>
<td><code>abortSubmission</code></td>
<td>abort the submission</td>
</tr>
<tr class="odd">
<td>Result</td>
<td><code>monitorSubmission</code></td>
<td>monitor the status of your workflow run</td>
</tr>
<tr class="even">
<td></td>
<td><code>listOutput</code></td>
<td>display the list of your workflow outputs</td>
</tr>
<tr class="odd">
<td></td>
<td><code>getOutput</code></td>
<td>download your output</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="google-cloud-sdk">Google Cloud SDK<a class="anchor" aria-label="anchor" href="#google-cloud-sdk"></a>
</h3>
<p>If you use RunTerraWorkflow within Terra’s RStudio, you don’t need extra authentication and gcloud SDK. If you intend to use this package from local machine, it requires that the gcloud SDK is installed, and that the billing account used by AnVIL can be authenticated with the user. These requirements are satisfied when using the AnVIL compute cloud. For local use, one must <a href="https://cloud.google.com/sdk/docs/install" class="external-link">install</a> the gcloud sdk.</p>
<p>Test the installation with <code><a href="https://rdrr.io/pkg/AnVIL/man/gcloud.html" class="external-link">AnVIL::gcloud_exists()</a></code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## gcloud_exists() should return TRUE</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnVIL/man/gcloud.html" class="external-link">gcloud_exists</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>If it returns <code>FALSE</code>, install the gcloud SDK following this script:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"rstudio/cloudml"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>cloudml<span class="sc">::</span><span class="fu">gcloud_install</span>()</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>gcloud auth login</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="do">## You can change the project using this script</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>gcloud config set project PROJECT_ID</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-in-this-vignette-microbiome-analysis">Example in this vignette: Microbiome analysis<a class="anchor" aria-label="anchor" href="#example-in-this-vignette-microbiome-analysis"></a>
</h3>
<p>In this vignette, we are running <a href="https://github.com/biobakery/biobakery_workflows" class="external-link">Whole Metagenome Shotgun (wmgx) analysis workflow</a> using <a href="https://github.com/biobakery/biobakery/wiki" class="external-link">bioBakery</a> tools. You can find the currently available workspaces using <code><a href="../reference/availableAnalysis.html">availableAnalysis()</a></code> function and the values under <code>analysis</code> column can be used for the analysis argument. For this vignette, we use <code>"bioBakery"</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/availableAnalysis.html">availableAnalysis</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;    analysis</span>
<span class="co">#&gt; 1 bioBakery</span>
<span class="co">#&gt; 2    salmon</span>
<span class="co">#&gt;                                                                                              description</span>
<span class="co">#&gt; 1                                                                    Microbiome analysis using bioBakery</span>
<span class="co">#&gt; 2 Trascript quantification from RNAseq using Salmon | Differential gene expression analysis using DESeq2</span>

<span class="va">analysis</span> <span class="op">&lt;-</span> <span class="st">"bioBakery"</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="section level3">
<h3 id="clone-workspace">Clone Workspace<a class="anchor" aria-label="anchor" href="#clone-workspace"></a>
</h3>
<p>First, you should clone the template workspace using <code>cloneWorkspace</code> function. Note that you need to provide a <strong>unique</strong> name for the cloned workspace through <code>workspaceName</code> argument.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">workspaceName</span> <span class="op">&lt;-</span> <span class="st">"biobakery_test"</span></code></pre></div>
<p>If you attempt to clone the template workspace using the existing workspaceName, you will get the below error message.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cloneWorkspace.html">cloneWorkspace</a></span><span class="op">(</span><span class="va">workspaceName</span>, <span class="va">analysis</span><span class="op">)</span>
<span class="co">#&gt; Error : 'avworkspace_clone' failed:</span>
<span class="co">#&gt;   Conflict (HTTP 409).</span>
<span class="co">#&gt;   Workspace waldronlab-terra-rstudio/biobakery_test already exists</span></code></pre></div>
<p>With the unique workspace name, you can successfully clone the workspace and the function will return the name of the cloned workspace.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cloneWorkspace.html">cloneWorkspace</a></span><span class="op">(</span>workspaceName <span class="op">=</span> <span class="st">"microbiome"</span>, <span class="va">analysis</span><span class="op">)</span>
<span class="co">#&gt; [1] "waldronlab-terra-rstudio/microbiome"</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-input">Prepare Input<a class="anchor" aria-label="anchor" href="#prepare-input"></a>
</h3>
<div class="section level4">
<h4 id="current-input">Current input<a class="anchor" aria-label="anchor" href="#current-input"></a>
</h4>
<p>You can review the current inputs using <code>currentInput</code> function. Below shows all the required and optional inputs for the workflow.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/currentInput.html">currentInput</a></span><span class="op">(</span><span class="va">workspaceName</span><span class="op">)</span>
<span class="co">#&gt; No encoding supplied: defaulting to UTF-8.</span>
<span class="co">#&gt; $workflowMTX.dataType</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.inputRead1Files</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/IBDMDB/ibdmdb_file_list.txt\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.MaxMemGB_QualityControlTasks</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.inputExtension</span>
<span class="co">#&gt; [1] "\".fastq.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.versionSpecifichumanDB</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/databases/kneaddata/Homo_sapiens_hg37_and_human_contamination_Bowtie2_v0.1.tar.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.versionSpecificUtilityMapping</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/databases/humann/full_mapping_v201901.tar.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.versionSpecificrrnaDB</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/databases/kneaddata/SILVA_128_LSUParc_SSUParc_ribosomal_RNA_v0.2.tar.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.customQCDB1</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.inputRead1Identifier</span>
<span class="co">#&gt; [1] "\"_R1\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.versionSpecifictrancriptDB</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/databases/kneaddata/Homo_sapiens_hg38_transcriptome_Bowtie2_v0.1.tar.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.preemptibleAttemptsOverride</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.inputRead2Identifier</span>
<span class="co">#&gt; [1] "\"_R2\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.ProjectName</span>
<span class="co">#&gt; [1] "\"ibdmdb_test\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.MaxMemGB_TaxonomicProfileTasks</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.versionSpecificChocophlan</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/databases/humann/full_chocophlan.v296_201901.tar.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.customQCDB2</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.bypassFunctionalProfiling</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.MaxMemGB_FunctionalProfileTasks</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.inputMetadataFile</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/IBDMDB/ibdmdb_demo_metadata.txt\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.versionSpecificUniRef90</span>
<span class="co">#&gt; [1] "\"gs://run_terra_workflow/databases/humann/uniref90_annotated_v201901.tar.gz\""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $workflowMTX.AdapterType</span>
<span class="co">#&gt; [1] "\"NexteraPE\""</span></code></pre></div>
<p><br></p>
<p><code><a href="https://rdrr.io/pkg/biobakeR/man/biobakery_currentInput.html" class="external-link">biobakeR::biobakery_currentInput</a></code> is a variation of <code>currentInput</code> function optimized for bioBakery workflow. It displays only the two major inputs - <code>inputListPath</code> and <code>inputFilePath</code>. These two types of files should be stored in Google bucket.</p>
<ol style="list-style-type: decimal">
<li>
<code>inputFilePath</code> file contains the paths to the input fastq files<br>
</li>
<li>
<code>inputListPath</code> file is a list of their paths as a txt file</li>
</ol>
<p>In other words, <code>inputFilePath</code> is the content of <code>inputListPath</code> - you save fastq files and provide their file paths in the txt file.</p>
<p>This vignette uses six fastq files for the test. Using preemptive instances, this demo set will cost about $5 to run.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">biobakeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biobakeR/man/biobakery_currentInput.html" class="external-link">biobakery_currentInput</a></span><span class="op">(</span><span class="va">workspaceName</span><span class="op">)</span>
<span class="co">#&gt; No encoding supplied: defaulting to UTF-8.</span>
<span class="co">#&gt; $inputListPath</span>
<span class="co">#&gt; [1] "gs://run_terra_workflow/IBDMDB/ibdmdb_file_list.txt"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $inputFilePath</span>
<span class="co">#&gt; [1] "gs://run_terra_workflow/IBDMDB/HSM7J4NY_R1.fastq.gz"</span>
<span class="co">#&gt; [2] "gs://run_terra_workflow/IBDMDB/HSMA33OT_R1.fastq.gz"</span>
<span class="co">#&gt; [3] "gs://run_terra_workflow/IBDMDB/MSM6J2QD_R1.fastq.gz"</span>
<span class="co">#&gt; [4] "gs://run_terra_workflow/IBDMDB/CSM9X23N_R1.fastq.gz"</span>
<span class="co">#&gt; [5] "gs://run_terra_workflow/IBDMDB/HSM6XRQY_R1.fastq.gz"</span>
<span class="co">#&gt; [6] "gs://run_terra_workflow/IBDMDB/HSMA33KE_R1.fastq.gz"</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="update-input">Update input<a class="anchor" aria-label="anchor" href="#update-input"></a>
</h4>
<p>Before launching the workflow, you should provide the correct input information using <code>updateInput</code> function. RunTerraWorkflow doesn’t support this function yet, but the bioBakery-specific function is available through <code><a href="https://rdrr.io/pkg/biobakeR/man/biobakery_updateInput.html" class="external-link">biobakeR::biobakery_updateInput</a></code>.</p>
<p>We use the six demo input files and their metadata.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">input</span> <span class="op">&lt;-</span> <span class="st">"gs://run_terra_workflow/IBDMDB/ibdmdb_file_list.txt"</span>
<span class="va">inputMeta</span> <span class="op">&lt;-</span> <span class="st">"gs://run_terra_workflow/IBDMDB/ibdmdb_demo_metadata.txt"</span>

<span class="fu">biobakeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biobakeR/man/biobakery_updateInput.html" class="external-link">biobakery_updateInput</a></span><span class="op">(</span><span class="va">workspaceName</span>,
                                InputRead1Files <span class="op">=</span> <span class="va">input</span>,
                                InputMetadataFile <span class="op">=</span> <span class="va">inputMeta</span>,
                                AdapterType <span class="op">=</span> <span class="st">"NexteraPE"</span>,
                                ProjectName <span class="op">=</span> <span class="st">"ibdmdb_test"</span>,
                                InputExtension <span class="op">=</span> <span class="st">".fastq.gz"</span>,
                                InputRead1Identifier <span class="op">=</span> <span class="st">"_R1"</span>,
                                InputRead2Identifier <span class="op">=</span> <span class="st">"_R2"</span><span class="op">)</span>
<span class="co">#&gt; No encoding supplied: defaulting to UTF-8.</span>
<span class="co">#&gt; [1] "Input information is succesfully updated."</span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="run-biobakery-workflow">Run bioBakery workflow<a class="anchor" aria-label="anchor" href="#run-biobakery-workflow"></a>
</h2>
<p>Once you clone the template workspace and update the input with your own data, you can launch the workflow using <code>launchWorkflow</code> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/launchWorkflow.html">launchWorkflow</a></span><span class="op">(</span><span class="va">workspaceName</span><span class="op">)</span>
<span class="co">#&gt; No encoding supplied: defaulting to UTF-8.</span>
<span class="co">#&gt; [1] "Workflow is succesfully launched."</span></code></pre></div>
<div class="section level3">
<h3 id="monitor-progress">Monitor Progress<a class="anchor" aria-label="anchor" href="#monitor-progress"></a>
</h3>
<p>The last three columns show the submission and the result status.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">submissions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/monitorSubmission.html">monitorSubmission</a></span><span class="op">(</span><span class="va">workspaceName</span><span class="op">)</span>
<span class="va">submissions</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 23 × 6</span></span>
<span class="co">#&gt;    submissionId            submitter submissionDate      status succeeded failed</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dttm&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 0ac4c1e4-b206-40a9-a4b… shbrief@… 2022-04-13 <span style="color: #949494;">16:38:36</span> Submi…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 888bb40b-afdd-4477-b69… shbrief@… 2022-04-13 <span style="color: #949494;">15:44:28</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> f6f79d38-825f-4bb9-89f… shbrief@… 2022-04-13 <span style="color: #949494;">00:57:27</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> eca13abb-a98d-4191-858… shbrief@… 2022-04-12 <span style="color: #949494;">22:17:04</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 154ecf88-a4d0-4859-8fd… shbrief@… 2022-04-12 <span style="color: #949494;">18:46:45</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 9f71497d-45da-4d89-908… shbrief@… 2022-04-12 <span style="color: #949494;">00:51:21</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> f158e487-a02a-4de6-924… shbrief@… 2022-04-12 <span style="color: #949494;">00:47:07</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 4ac530ad-11cf-43b8-91a… shbrief@… 2022-04-12 <span style="color: #949494;">00:43:55</span> Done           1      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 96071a85-03be-4fe2-8d8… shbrief@… 2022-04-12 <span style="color: #949494;">00:42:43</span> Done           1      0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 83d92e35-4540-4133-957… shbrief@… 2022-04-12 <span style="color: #949494;">00:23:21</span> Abort…         0      0</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 13 more rows</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="abort-submission">Abort submission<a class="anchor" aria-label="anchor" href="#abort-submission"></a>
</h3>
<p>To abort the most recently submitted job, you don’t need to specify <code>submissionId</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/abortSubmission.html">abortSubmission</a></span><span class="op">(</span><span class="va">workspaceName</span><span class="op">)</span>
<span class="co">#&gt; Status of the submitted job (submissionId: 0ac4c1e4-b206-40a9-a4be-59ca01a925c0)</span>
<span class="co">#&gt; [1] "Workflow is succesfully aborted."</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="result">Result<a class="anchor" aria-label="anchor" href="#result"></a>
</h2>
<div class="section level3">
<h3 id="list-outputs">List Outputs<a class="anchor" aria-label="anchor" href="#list-outputs"></a>
</h3>
<p>You can check all the output files from the most recently succeeded submission. If you specify the <code>submissionId</code> argument, you can get the output files of that specific submission. Output files can also be subset using <code>keyword</code> argument.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listOutput.html">listOutput</a></span><span class="op">(</span><span class="va">workspaceName</span><span class="op">)</span>
<span class="co">#&gt; Outputs are from the submissionId 4ac530ad-11cf-43b8-91a0-350bf91a2295</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 193 × 5</span></span>
<span class="co">#&gt;    file                                    workflow    task          type  path </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> humann_ecs_relab_counts.tsv             workflowMTX call-CountRe… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> humann_genefamilies_relab_counts.tsv    workflowMTX call-CountRe… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> humann_pathabundance_relab_counts.tsv   workflowMTX call-CountRe… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> humann_read_and_species_count_table.tsv workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CSM9X23N_bowtie2_unaligned.fa           workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CSM9X23N_diamond_unaligned.fa           workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CSM9X23N_genefamilies.tsv               workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CSM9X23N_pathabundance.tsv              workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CSM9X23N_pathcoverage.tsv               workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> HSM6XRQY_bowtie2_unaligned.fa           workflowMTX call-Functio… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 183 more rows</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listOutput.html">listOutput</a></span><span class="op">(</span><span class="va">workspaceName</span>, keyword <span class="op">=</span> <span class="st">"humann"</span><span class="op">)</span>
<span class="co">#&gt; Outputs are from the submissionId 4ac530ad-11cf-43b8-91a0-350bf91a2295</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 5</span></span>
<span class="co">#&gt;   file                                    workflow    task           type  path </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> humann_ecs_relab_counts.tsv             workflowMTX call-CountRel… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> humann_genefamilies_relab_counts.tsv    workflowMTX call-CountRel… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> humann_pathabundance_relab_counts.tsv   workflowMTX call-CountRel… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> humann_read_and_species_count_table.tsv workflowMTX call-Function… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> humann_feature_counts.tsv               workflowMTX call-JoinFeat… outp… gs:/…</span></code></pre></div>
<p>If your outputs include tsv files, you can check the head of those files using <code>tableHead</code> function without downloading them.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tableHead.html">tableHead</a></span><span class="op">(</span><span class="st">"HSM7J4NY_genefamilies_relab.tsv"</span>, n <span class="op">=</span> <span class="fl">6</span>, <span class="va">workspaceName</span><span class="op">)</span>   
<span class="co">#&gt; Below outputs are from the submissionId 4ac530ad-11cf-43b8-91a0-350bf91a2295</span>
<span class="co">#&gt;                                                                      V1</span>
<span class="co">#&gt; 1                                                       UniRef90_B7B7I3</span>
<span class="co">#&gt; 2          UniRef90_B7B7I3|g__Parabacteroides.s__Parabacteroides_merdae</span>
<span class="co">#&gt; 3                                          UniRef90_B7B7I3|unclassified</span>
<span class="co">#&gt; 4                                                       UniRef90_F3QJ09</span>
<span class="co">#&gt; 5 UniRef90_F3QJ09|g__Parasutterella.s__Parasutterella_excrementihominis</span>
<span class="co">#&gt; 6                                                   UniRef90_A0A374V756</span>
<span class="co">#&gt;            V2</span>
<span class="co">#&gt; 1 2.25732e-01</span>
<span class="co">#&gt; 2 2.25683e-01</span>
<span class="co">#&gt; 3 4.95507e-05</span>
<span class="co">#&gt; 4 1.39816e-01</span>
<span class="co">#&gt; 5 1.39816e-01</span>
<span class="co">#&gt; 6 3.66683e-02</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-outputs">Get Outputs<a class="anchor" aria-label="anchor" href="#get-outputs"></a>
</h3>
<p><code>keyword</code> argument takes a character string containing a regular expression. In the example below, we check all the <code>.tsv</code> outputs of the sample, “HSM7J4NY”.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/listOutput.html">listOutput</a></span><span class="op">(</span><span class="va">workspaceName</span>, keyword <span class="op">=</span> <span class="st">"HSM7J4NY.*.tsv"</span><span class="op">)</span>
<span class="co">#&gt; Outputs are from the submissionId 4ac530ad-11cf-43b8-91a0-350bf91a2295</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 9 × 5</span></span>
<span class="co">#&gt;   file                             workflow    task                  type  path </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> HSM7J4NY_genefamilies.tsv        workflowMTX call-FunctionalProfi… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> HSM7J4NY_pathabundance.tsv       workflowMTX call-FunctionalProfi… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> HSM7J4NY_pathcoverage.tsv        workflowMTX call-FunctionalProfi… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> HSM7J4NY_ecs.tsv                 workflowMTX call-RegroupECs       outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> HSM7J4NY_kos.tsv                 workflowMTX call-RegroupKOs       outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> HSM7J4NY_ecs_relab.tsv           workflowMTX call-RenormTableECs   outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">7</span> HSM7J4NY_genefamilies_relab.tsv  workflowMTX call-RenormTableGenes outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">8</span> HSM7J4NY_pathabundance_relab.tsv workflowMTX call-RenormTablePath… outp… gs:/…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">9</span> HSM7J4NY.tsv                     workflowMTX call-TaxonomicProfile outp… gs:/…</span></code></pre></div>
<p>You can download any file using <code>getOutput</code> function. Here, we narrow down the download to HSM7J4NY sample’s <code>.tsv</code> output files.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getOutput.html">getOutput</a></span><span class="op">(</span><span class="va">workspaceName</span>, keyword <span class="op">=</span> <span class="st">"HSM7J4NY.*.tsv"</span>, dest_dir <span class="op">=</span> <span class="va">HSM7J4NY_dir</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">HSM7J4NY_dir</span><span class="op">)</span>
<span class="co">#&gt; [1] "HSM7J4NY_ecs_relab.tsv"           "HSM7J4NY_ecs.tsv"                </span>
<span class="co">#&gt; [3] "HSM7J4NY_genefamilies_relab.tsv"  "HSM7J4NY_genefamilies.tsv"       </span>
<span class="co">#&gt; [5] "HSM7J4NY_kos.tsv"                 "HSM7J4NY_pathabundance_relab.tsv"</span>
<span class="co">#&gt; [7] "HSM7J4NY_pathabundance.tsv"       "HSM7J4NY_pathcoverage.tsv"       </span>
<span class="co">#&gt; [9] "HSM7J4NY.tsv"</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: aarch64-apple-darwin20 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Monterey 12.2.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] AnVIL_1.7.14           dplyr_1.0.8            RunTerraWorkflow_0.3.2</span>
<span class="co">#&gt; [4] BiocStyle_2.22.0      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] tidyselect_1.1.2     xfun_0.30            bslib_0.3.1         </span>
<span class="co">#&gt;  [4] purrr_0.3.4          vctrs_0.4.0          generics_0.1.2      </span>
<span class="co">#&gt;  [7] htmltools_0.5.2      yaml_2.3.5           utf8_1.2.2          </span>
<span class="co">#&gt; [10] rlang_1.0.2          pkgdown_2.0.2        jquerylib_0.1.4     </span>
<span class="co">#&gt; [13] pillar_1.7.0         rapiclient_0.1.3     glue_1.6.2          </span>
<span class="co">#&gt; [16] DBI_1.1.2            lambda.r_1.2.4       lifecycle_1.0.1     </span>
<span class="co">#&gt; [19] stringr_1.4.0        futile.logger_1.4.3  ragg_1.2.2          </span>
<span class="co">#&gt; [22] memoise_2.0.1        evaluate_0.15        knitr_1.38          </span>
<span class="co">#&gt; [25] fastmap_1.1.0        parallel_4.1.2       curl_4.3.2          </span>
<span class="co">#&gt; [28] fansi_1.0.3          formatR_1.12         BiocManager_1.30.16 </span>
<span class="co">#&gt; [31] cachem_1.0.6         biobakeR_0.4.0       desc_1.4.1          </span>
<span class="co">#&gt; [34] jsonlite_1.8.0       systemfonts_1.0.4    fs_1.5.2            </span>
<span class="co">#&gt; [37] textshaping_0.3.6    digest_0.6.29        stringi_1.7.6       </span>
<span class="co">#&gt; [40] bookdown_0.25        rprojroot_2.0.3      cli_3.2.0           </span>
<span class="co">#&gt; [43] tools_4.1.2          magrittr_2.0.3       sass_0.4.1          </span>
<span class="co">#&gt; [46] tibble_3.1.6         futile.options_1.0.1 crayon_1.5.1        </span>
<span class="co">#&gt; [49] tidyr_1.2.0          pkgconfig_2.0.3      ellipsis_0.3.2      </span>
<span class="co">#&gt; [52] assertthat_0.2.1     rmarkdown_2.13       httr_1.4.2          </span>
<span class="co">#&gt; [55] rstudioapi_0.13      R6_2.5.1             compiler_4.1.2</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Sehyun Oh.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
